src/Internal/Contract/Wallet.purs

```diff
@@ -251,3 +279,51 @@ getWalletUtxos = do
   toUtxoMap :: Array TransactionUnspentOutput -> UtxoMap
   toUtxoMap = Map.fromFoldable <<< map
     (unwrap >>> \({ input, output }) -> input /\ output)
+
+ownDrepPubKey :: Contract PublicKey
+ownDrepPubKey =
+  withWallet do
+    actionBasedOnWallet _.getPubDrepKey
+      ( \kw -> do
+          drepKey <- liftAff $ liftedM
+            (error "ownDrepPubKey: Unable to get KeyWallet DRep key")
+            (getPrivateDrepKey kw)
+          pure $ PrivateKey.toPublicKey $ unwrap drepKey
+      )
+
+ownDrepPubKeyHash :: Contract Ed25519KeyHash
+ownDrepPubKeyHash =
+  withWallet do
+    actionBasedOnWallet (map PublicKey.hash <<< _.getPubDrepKey)
+      ( \kw -> do
+          drepKey <- liftAff $ liftedM
+            (error "ownDrepPubKeyHash: Unable to get KeyWallet DRep key")
+            (getPrivateDrepKey kw)
+          pure $ PublicKey.hash $ PrivateKey.toPublicKey $
+            unwrap drepKey
+      )
+
+ownRegisteredPubStakeKeys :: Contract (Array PublicKey)
+ownRegisteredPubStakeKeys =
+  withWallet do
+    actionBasedOnWallet _.getRegisteredPubStakeKeys
+      (kwPubStakeKeys "ownRegisteredPubStakeKeys")
+
+ownUnregisteredPubStakeKeys :: Contract (Array PublicKey)
+ownUnregisteredPubStakeKeys =
+  withWallet do
+    actionBasedOnWallet _.getUnregisteredPubStakeKeys
+      (kwPubStakeKeys "ownUnregisteredPubStakeKeys")
+
+kwPubStakeKeys :: String -> KeyWallet -> Contract (Array PublicKey)
+kwPubStakeKeys funName kw = do
+  logWarn' $ funName <>
+    " via KeyWallet: KeyWallet does not distinguish between \
```

Comment:
I think the spec implies that if it's impossible to know, all keys should be considered unregistered:

> api.cip95.getUnregisteredPubStakeKeys
> If the wallet does not know the registration status of it's stake keys then it should return them as part of this call

Reply:
That's a great point, I'll update the implementation

Reply:
Addressed in https://github.com/Plutonomicon/cardano-transaction-lib/pull/1638/commits/cbf1fa857e45d1a8717ddfccb5ebe8df82f36465